text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,533 @@
+unpacked npm package size to 234 MB, at the time of writing. This feature
+reduces the size of the package to under 100 MB.
+
+## Working Backwards
+
+* **CHANGELOG**:
+
+feat(aws-cdk-lib): Reduce aws-cdk-lib package size.
+
+‼️ If you use any of the following Constructs in a network-restricted
+environment, you might encounter problems upgrading to this version. ‼️
+
+* lambda_layer_kubectl.KubectlLayer
+* aws_eks.KubectlProvider
+* aws_eks.HelmChart
+* aws_eks.KubernetesManifest
+* aws_eks.KubernetesObjectValue
+* aws_eks.KubernetesPatch
+* lambda_layer_awscli.AwsCliLayer
+* aws_s3_deployment.BucketDeploymentß
+* aws_stepfunctions_tasks.EmrContainersStartJobRun
+* lambda_layer_node_proxy_agent.NodeProxyAgentLayer
+* aws_eks.FargateProfile
+* aws_eks.Cluster
+
+Please see [#1234](https://github.com/aws/aws-cdk/issues/1234) for more details,
+and comment on the issue if you run into problems.
+
+* **CLI Notices**:
+
+We can deliver a notice to customers based on what specific constructs they are
+using. We will define separate notices for each different scenario.
+
+Title: (aws-cdk-lib): upcoming change in packaging structure might require
+action to upgrade
+
+Body: We’ve identified that you are using at least one of the [List of
+Constructs] Constructs. If you are running your aws-cdk commands in an
+environment that does not have access to npm, then you will need to make a
+change to make version 2.x of [npm packages] available in your environment. See
+[#1234](https://github.com/aws/aws-cdk/issues/1234) for details and
+instructions.
+
+| List of Constructs                                                                        | npm packages                                                                                            |
+| ----------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
+| KubectlLayer                                                                              | @aws-cdk/lambda-layer-kubectl                                                                           |
+| KubectlProvider, HelmChart, KubernetesManifest, KubernetesObjectValue, or KubernetesPatch | @aws-cdk/lambda-layer-kubectl and @aws-cdk/lambda-layer-awscli                                          |
+| AwsCliLayer, BucketDeployment, or EmrContainersStartJobRun                                | @aws-cdk/lambda-layer-awscli                                                                            |
+| NodeProxyAgentLayer, or FargateProfile                                                    | @aws-cdk/lambda-layer-node-proxy-agent                                                                  |
+| aws_eks.Cluster                                                                           | @aws-cdk/lambda-layer-kubectl, @aws-cdk/lambda-layer-awscli, and @aws-cdk/lambda-layer-node-proxy-agent |
```

Comment:
Can we use the module prefix everywhere in the table? I think it will make it easier to see on first glance. And maybe the full prefix? (`aws-cdk-lib/aws_eks`).

Maybe put each construct on its own line?

Reply:
Good idea, done :)